protocol/v2/qbft/storage/ibft_store.go

```diff
@@ -32,17 +36,63 @@ type ParticipantsRangeEntry struct {
 	Identifier convert.MessageID
 }
 
+type Quorum struct {
+	Signers     []spectypes.OperatorID
+	Committee   []spectypes.OperatorID
+	ValidatorPK spectypes.ValidatorPK // optional payload
+}
```

Comment:
It doesn't seem like `qbftstorage` package is the best place to define `Quorum` struct in, wouldn't `v2/validator` be a better place for it ?

For example, `quorum.ValidatorPK` is never even accessed/needed inside of `qbftstorage` package, this kind of implies `Quorum` abstraction isn't the best fit for what we are trying to do here, and instead feels like going with a dumb "bitmask" + decoder/encoder for it is just simpler way to do it (might not be as elegant, but probably is more straightforward to debug and deal with).

Reply:
I agree that the structure is not the best, but moving this would cause cyclic import because the `QBFTStore` interface uses this type. I don't like the current type relations between these packages but it will probably be easier to resolve them later. 

I don't like the existence of `ValidatorPK` inside `Quorum` either, I just didn't want to create a new type for it, but I likely should because we already have `validatorIndexAndRoot`. Done